### PR TITLE
feat(FullEnd2EndTest): initial test script integrated in to github workflow

### PR DESCRIPTION
**Pull request by Aaron-Ritter**
This pull request aims to deliver automated Full E2E testing workflows with multiple Android and FusionAuth versions. The following features and changes have been integrated into this branch:

1. Add Full End 2 End Test with espresso and uiautomator testing: login, request refresh token, logout
2. Add FusionAuth Docker Compose configuration for the multiple previous and current versions
3. Add e2e-test-fusionauth-latest.yml Workflow to automatically run the Full End 2 End Test on Android 30, 31, 33, and 34
4. Add e2e-test-fusionauth-supported.yml Workflow to manually run the Full End 2 End Test on Android 33 and all currently supported FusionAuth versions
5. The tests make sure of running the emulator, pre-configure chrome, running the Full End 2 End Test in a given matrix, recording the failure in case the Test fails, and uploading it as a artifact.

### CODE REVIEW

File: app/src/androidTest/java/io/fusionauth/sdk/FullEnd2EndTest.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        Intents.release()
+    }
+
+    companion object {
+        private val logger = Logger.getLogger(FullEnd2EndTest::class.java.name)
+
+        private const val USERNAME = "richard@example.com"
+        private const val PASSWORD = "password"
+        private const val TIMEOUT_MILLIS = 10_000L
```
[the remark this answers is not in the document]
Comment (Aaron-Ritter) [reply app/src/androidTest/java/io/fusionauth/sdk/FullEnd2EndTest.kt:127]: The password is used only in Full End 2 End Test and needs to be visible for developers to understand the testing process. This is the same password as defined in the fusionauth kickstart.

**Comment by Aaron-Ritter** [comment]
Still needs a bit fine tuning, as the test seems to fail, according to the uploaded recording, because the keyboard sometimes overlaps the input fields. 
- https://stackoverflow.com/questions/17223305/suppress-keyboard-after-setting-text-with-android-uiautomator
- https://stackoverflow.com/questions/28311511/detect-or-suppress-keyboard-in-uiautomator-tests

File: app/src/androidTest/java/io/fusionauth/sdk/FullEnd2EndTest.kt [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+        Intents.release()
+    }
+
+    companion object {
+        private val logger = Logger.getLogger(FullEnd2EndTest::class.java.name)
+
+        private const val USERNAME = "richard@example.com"
+        private const val PASSWORD = "password"
+        private const val USERNAME2 = "gilfoyle@example.com"
```
[the remark this answers is not in the document]
Comment (Aaron-Ritter) [reply app/src/androidTest/java/io/fusionauth/sdk/FullEnd2EndTest.kt:191]: Only used for End 2 End testing, and needs to be known for developers to run and modify the workflow.

**Comment by Aaron-Ritter** [comment]
ready for review

**Review by ColinFrick — APPROVED** [review]
LGTM

**Pull request merged**